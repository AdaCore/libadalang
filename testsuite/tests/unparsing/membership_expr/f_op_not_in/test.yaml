driver: unparser
rule: subp_body
description: |
  Test that the `f_membership_exprs` field is correctly indented based on
  `f_expr` when `f_op` is an `OpNotIn` node.
