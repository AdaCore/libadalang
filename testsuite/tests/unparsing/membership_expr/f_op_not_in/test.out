procedure Foo is
begin
   if Parent.Parent.Kind not in
        Ada_Begin_Block_Range
        | Ada_Decl_Block_Range
        | Bar
     and then Parent_Parent.Parent_Parent.Parent_Parent.Parent_Kind not in
                Ada_Begin_Block_Range
                | Ada_Decl_Block_Range
                | Bar
     and then Parent_Parent
                .Parent_Parent
                .Parent_Parent
                .Parent_Parent
                .Parent_Kind
              not in Ada_Begin_Block_Range
                     | Ada_Decl_Block_Range
                     | Bar
   then
      Current_Indentation := @ + Indentation;
   end if;
end Foo;
